server/test/handlers/availability.test.js

```diff
@@ -4,40 +4,197 @@ describe('testing fetching of worker availability functionality', () => {
+
+        // Assert
+        expect(availableTimes).toBe(null);
+    });
+
+    //TEST 6
 
-    test('initial setup of worker availability fetching', async() => {
+    test('rejection of empty string value for required workerId parameter', async () => {
+        // Arrange
+        const workerId = '';
+        const schoolId = 1;
+        const startTime = '8:00:00';
+        const endTime = '8:30:00';
+
+        // Act
+        const availableTimes = await availabilityHandler.getAvailabilityDetails(workerId, schoolId, startTime, endTime);
+
+        // Assert
+        expect(availableTimes).toBe(null);
+    });
+
+    //TEST 7
+    test('rejection of empty string value for required schoolId parameter', async () => {
+        // Arrange
+        const workerId = 8000000;
+        const schoolId = '';
+        const startTime = '8:00:00';
+        const endTime = '8:30:00';
+
+        // Act
+        const availableTimes = await availabilityHandler.getAvailabilityDetails(workerId, schoolId, startTime, endTime);
+
+        // Assert
+        expect(availableTimes).toBe(null);
+    });
+    //TEST 8
+
+    test('rejection of empty string value for required startTime parameter', async () => {
+        // Arrange
+        const workerId = 8000000;
+        const schoolId = 1;
+        const startTime = '';
+        const endTime = '8:30:00';
+
+        // Act
+        const availableTimes = await availabilityHandler.getAvailabilityDetails(workerId, schoolId, startTime, endTime);
+
+        // Assert
+        expect(availableTimes).toBe(null);
+    });
+    //TEST 9
+    test('rejection of empty string value for required endTime parameter', async () => {
         // Arrange
         const workerId = 8000000;
         const schoolId = 1;
-        const startTime = '2020-10-20';
-        const endTime = '2020-10-20';
-
-        const testAvailableTimes = [
-            {
-                workerTimeslotId: 1,
-                startTime: '08:00:00', 
-                endTime: '08:30:00',
-                date: '2020-10-20'
-            },
-            {
-                workerTimeslotId: 2,
-                startTime: '08:30:00', 
-                endTime: '09:00:00',
-                date: '2020-10-20'
-            },
-            {
-                workerTimeslotId: 3,
-                startTime: '09:00:00', 
-                endTime: '09:30:00',
-                date: '2020-10-20'
-            }
-        ];
-
-        // Act
-        const availableTimes = await availabilityHandler.getWorkerAvailability(workerId, schoolId, startTime, endTime);
-
-        // Assert
-        expect(availableTimes).toMatchObject(testAvailableTimes);
+        const startTime = '8:00:00';
+        const endTime = '';
+
+        // Act
+        const availableTimes = await availabilityHandler.getAvailabilityDetails(workerId, schoolId, startTime, endTime);
+
+        // Assert
+        expect(availableTimes).toBe(null);
     });
+    /*
+    //TEST 10
```

Comment:
Hi Nancy, 
This appears to be a valid test and I was wondering if it was possible to explain why it was commented out,

Reply:
Hi Amy,

This test is now removed to avoid confusion. The condition being checked by the test is taken care of in line 6 of server/models/handlers/availability.js :)